src/LineChart/components/LineChartContainer.ts

```diff
@@ -0,0 +1,236 @@
+    }
+
+    public static validateProps(props: LineChartContainerProps): string {
+        let errorMessage = "";
+        if (props.sourceType === "microflow" && !props.dataSourceMicroflow) {
+            errorMessage += ` data source type is set to 'Microflow' but 'Source microflow' is missing \n`;
+        }
+
+        return errorMessage && `Configuration error :\n\n ${errorMessage}`;
+    }
+
+    private resetSubscriptions(mxObject?: mendix.lib.MxObject) {
+        this.subscriptionHandles.forEach(mx.data.unsubscribe);
+        this.subscriptionHandles = [];
+
+        if (mxObject) {
+            this.subscriptionHandles.push(mx.data.subscribe({
+                callback: this.handleSubscription,
+                guid: mxObject.getGuid()
+            }));
+        }
+    }
+
+    private handleSubscription() {
+        this.fetchData(this.props.mxObject);
+    }
+
+    private fetchData(mxObject?: mendix.lib.MxObject) {
+        const { seriesEntity } = this.props;
+        if (mxObject && this.props.seriesEntity) {
+            if (this.props.sourceType === "xpath") {
+                const constraint = this.props.entityConstraint
+                    ? this.props.entityConstraint.replace("[%CurrentObject%]", mxObject.getGuid())
+                    : "";
+                const entityName = seriesEntity.indexOf("/") > -1
+                    ? seriesEntity.split("/")[seriesEntity.split("/").length - 1]
+                    : seriesEntity;
+                const XPath = "//" + entityName + constraint;
+                this.fetchByXPath(XPath);
+            } else if (this.props.sourceType === "microflow" && this.props.dataSourceMicroflow) {
+                this.fetchByMicroflow(mxObject.getGuid());
+            }
+
+        }
+    }
+
+    private fetchByXPath(xpath: string) {
+        window.mx.data.get({
+            callback: mxObjects => this.fetchDataFromSeries(mxObjects),
+            error: error => this.setState({
+                alertMessage: `An error occurred while retrieving data via XPath (${xpath}): ${error}`,
+                data: []
+            }),
+            xpath
+        });
+    }
+
+    private fetchByMicroflow(guid: string) {
+        const actionname = this.props.dataSourceMicroflow;
+        mx.ui.action(actionname, {
+            callback: mxObjects => {
+                const series = mxObjects as mendix.lib.MxObject[];
+                this.fetchDataFromSeries(series);
+            },
+            error: error => this.setState({
+                alertMessage: `Error while retrieving microflow data ${actionname}: ${error.message}`,
+                data: []
+            }),
+            params: {
+                applyto: "selection",
+                guids: [ guid ]
+            }
+        });
+    }
+
+    private fetchDataFromSeries(series: mendix.lib.MxObject[]) {
+        const seriesCount = series.length;
+        series.forEach((object, index) => {
+            const data: DataProps = {
```

Comment:
No need for an object here. Just create individual variables for whatever u need. Unless you're passing in the entire data variable somewhere.